scipy/ndimage/tests/test_filters.py

```diff
@@ -2016,6 +2016,38 @@ def test_rank15(self, dtype, xp):
                                      origin=[-1, 0])
         assert_array_almost_equal(expected, output)
 
+    def test_rank16(self):
+        array = [3, 2, 5, 1, 4]
+        expected = np.asarray([3, 3, 2, 4, 4])
+        output = ndimage.rank_filter(array, -2, size=3)
+        assert_array_almost_equal(expected, output)
```

Comment:
```suggestion
        xp_assert_equal(expected, output)
```
since you're comparing integers for exact equality

Reply:
done